src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int RATE = 3;
+    private static final int HOURS = 2;
+    private static final int NAME = 1;
+    private static final int DATE = 0;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder();
+        builder.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+        for (int i = 0, salary = 0; i < names.length; i++, salary = 0) {
+            for (int dataElement = 0; dataElement < data.length; dataElement++) {
+                String[] dataArray = data[dataElement].split(" ");
+                if (names[i].equals(dataArray[NAME])
+                        && LocalDate.parse(dataArray[DATE], DATE_FORMAT)
+                        .isAfter(LocalDate.parse(dateFrom, DATE_FORMAT))
+                        && (LocalDate.parse(dataArray[DATE], DATE_FORMAT))
+                        .isBefore((LocalDate.parse(dateTo, DATE_FORMAT)).plusDays(1))) {
```

Comment:
```suggestion
                        && LocalDate.parse(dataArray[DATE], DATE_FORMAT)
                        .isAfter(LocalDate.parse(dateFrom, DATE_FORMAT))
                        && (LocalDate.parse(dataArray[DATE], DATE_FORMAT))
                        .isBefore((LocalDate.parse(dateTo, DATE_FORMAT)))) {
```

Reply:
example with [isEqual](https://www.geeksforgeeks.org/localdate-isequal-method-in-java-with-examples/#:~:text=The%20isEqual()%20method%20of,the%20specified%20date%20or%20not.&text=Parameter%3A%20This%20method%20accept%20a,equal%20to%20the%20specified%20date.)